pages/write/[[...roomId]].js

```diff
@@ -166,132 +141,199 @@ export default function Write() {
-        const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_';
-        const charactersLength = characters.length;
-        for (let i = 0; i < 20; i++) {
-            string += characters.charAt(Math.floor(Math.random() * charactersLength));
-        }
-        const link = getConfig().publicRuntimeConfig.authProviders.write.baseUrl + '/' + string;
-        await createWriteRoom(link);
-        setLoading(false);
-    };
+        return room;
+    }, [auth, matrix, matrixClient, serviceSpaceId, write]);
+
+    const ActionNewAnonymousPad = () => {
+        const [padName, setPadName] = useState('');
+        const [loading, setLoading] = useState(false);
+
+        const createAnonymousPad = async () => {
+            setLoading(true);
+            let string = '';
+            const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_';
+            const charactersLength = characters.length;
+            for (let i = 0; i < 20; i++) {
+                string += characters.charAt(Math.floor(Math.random() * charactersLength));
+            }
+            const link = getConfig().publicRuntimeConfig.authProviders.write.baseUrl + '/' + string;
+            const roomId = await createWriteRoom(link, padName);
+            router.push(`/write/${roomId}`);
+
+            setLoading(false);
+            setPadName('');
+            // in order to close the service sub menu we need to set the toggle to true and reset it to false so it works again the next time around
+            setCloseSubemnuToggle(true);
+            setCloseSubemnuToggle(false);
+        };
 
-    const createPasswordPad = async () => {
-        setLoading(true);
-        const padId = await write.createPad(newPadName, 'private', password);
-        const link = getConfig().publicRuntimeConfig.authProviders.write.baseUrl + '/' + padId;
-        await createWriteRoom(link);
-        setLoading(false);
+        return (
+            <Form onSubmit={(e) => { e.preventDefault(); createAnonymousPad(padName); }}>
+                <input type="text" placeholder={t('pad name')} value={padName} onChange={(e) => setPadName(e.target.value)} />
+                <button type="submit" disabled={!padName}>{ loading ? <LoadingSpinnerInline inverted /> : t('Create pad') }</button>
+            </Form>);
     };
 
-    const createAuthoredPad = async () => {
-        setLoading(true);
-        const padId = await write.createPad(newPadName, 'public').catch((err) => {
-            console.log(err);
-        });
-        if (!padId) {
+    const ActionExistingPad = () => {
+        const [padName, setPadName] = useState('');
+        const [padLink, setPadLink] = useState('');
+        const [validLink, setValidLink] = useState('undefined');
+        const [loading, setLoading] = useState(false);
+
+        const handleExistingPad = (e) => {
+            if (getConfig().publicRuntimeConfig.authProviders.write.bypassUrlValidation) {
```

Comment:
A bit confused... if `bypassUrlValidation` is true... why do we check if the URL is valid then? Is this supposed to be the other way around? Can you add some commenting and more clear variable names on how this is intended to work?

Reply:
I think I understood... maybe a better variable name for the config variable would be something like "`allowThirdPartyPads`" oder "`allowPadsFromOtherSites`"?

Reply:
we can change the name to make it clearer. but for now (unless we change the validation) literally any link can be saved, though. there's no check if its an actual etherpad link or not.